frontend/src/core/ai/context/registry.ts

```diff
@@ -0,0 +1,190 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import type { Completion } from "@codemirror/autocomplete";
+import type { Resource } from "@marimo-team/codemirror-mcp";
+import type { TypedString } from "@/utils/typed";
+
+/**
+ * Unique identifier for a context item in the format "type:id"
+ * e.g., "variable://my_var", "data://users", "file://config.py"
+ */
+export type ContextLocatorId = TypedString<"ContextLocatorId">;
+
+/**
+ * Base interface for context items that can be mentioned in AI prompts
+ */
+export interface AIContextItem extends Resource<Record<string, unknown>> {
+  data: Record<string, unknown>;
+}
+
+/**
+ * Abstract base class for AI context providers with shared implementation
+ */
+export abstract class AIContextProvider<
+  T extends AIContextItem = AIContextItem,
+> {
+  /** Human-readable title for this context type */
+  abstract readonly title: string;
+
+  /** Prefix used for mentions (e.g., "@" for variables, "#" for files) */
+  abstract readonly mentionPrefix: string;
```

Comment:
Should it be one character? Should we but a restriction on it with a more precise type "@" | "#"?

Reply:
Yea that makes sense